crates/core/src/network/handler/fetch.rs

```diff
@@ -2,30 +2,40 @@ use crate::{
     action::{Action, ActionWrapper},
     context::Context,
     instance::dispatch_action,
-    network::handler::{get_content_aspect, get_meta_aspects},
+    network::handler::{
+        get_content_aspect, get_meta_aspects_from_chain, get_meta_aspects_from_dht_eav,
+    },
 };
+use holochain_core_types::network::entry_aspect::EntryAspect;
 use lib3h_protocol::data_types::FetchEntryData;
-use std::sync::Arc;
+use std::{collections::HashSet, sync::Arc};
 
 /// The network has requested a DHT entry from us.
 /// Lets try to get it and trigger a response.
 pub fn handle_fetch_entry(get_dht_data: FetchEntryData, context: Arc<Context>) {
     let address = get_dht_data.entry_address.clone();
-    let mut aspects = vec![];
+    let mut aspects: HashSet<EntryAspect> = HashSet::new();
 
     // XXX: NB: we seem to be ignoring aspect_address_list and just attempting to get all aspects.
     // Is that right?
 
     match get_content_aspect(&address, context.clone()) {
         Ok(content_aspect) => {
-            aspects.push(content_aspect);
-            match get_meta_aspects(&address, context.clone()) {
-                Ok(mut meta_aspects) => aspects.append(&mut meta_aspects),
-                Err(get_meta_error) => {
-                    log_error!(context, "net/handle_fetch_entry: Error getting meta aspects for entry ({:?}), error: {:?}",
-                        address,
-                        get_meta_error,
-                    );
+            aspects.insert(content_aspect);
+            for result in &[
+                get_meta_aspects_from_chain(&address, context.clone()),
+                get_meta_aspects_from_dht_eav(&address, context.clone()),
```

Comment:
This is where we used to only get meta aspects from the DHT/EAV (the old fn name `get_meta_aspects()` implies that we were not thinking about getting it from the chain at all). Now we ask both sources for meta aspects for the given entry address and merge the results.